Mathlib/RingTheory/NonZeroDivisors.lean

```diff
@@ -37,6 +35,19 @@ def nonZeroDivisors (R : Type*) [MonoidWithZero R] : Submonoid R where
 /-- The notation for the submonoid of non-zerodivisors. -/
 scoped[nonZeroDivisors] notation:9000 R "⁰" => nonZeroDivisors R
 
+/-- Let `R` be a monoid with zero and `M` an additive monoid with an `R`-action, then the collection
+of non-zero smul-divisors forms a submonoid. These elements are also called `M`-regular.-/
+def nonZeroSMulDivisors (R : Type*) [MonoidWithZero R] (M : Type _) [AddMonoid M] [MulAction R M] :
+    Submonoid R where
+  carrier := { r | ∀ m : M, r • m = 0 → m = 0}
+  one_mem' m h := (one_smul R m) ▸ h
+  mul_mem' {r₁ r₂} h₁ h₂ m H := h₂ _ $ h₁ _ $ mul_smul r₁ r₂ m ▸ H
+
+/-- The notation for the submonoid of non-zero smul-divisors. -/
+scoped[nonZeroSMulDivisors] notation:9000 R "⁰[" M "]" => nonZeroSMulDivisors R M
```

Comment:
Please mention this in the module docstring

Reply:
done